### ISSUE TITLE
How to disable the internal keyboard while using external one? (MacOS 10.12.1)

### ISSUE THREAD

**Issue by 0xABCDEF** [tekezo/Karabiner-Elements#334]
Thanks to Karabiner that I can put my external keyboard on Macbook's build-in keyboard.
Now I turn to Karabiner-Elements for working on MacOS 10.12.1 (Serria).
But I didn't find out how to disable the build-in keyboard.
Please let me know if there is any other available solution.

**Comment by nmqanh** [comment]
I think they are still working on that.
This issue is available here (with some temporary solution inside)
https://github.com/tekezo/Karabiner-Elements/issues/95

**Comment by jdelStrother** [comment]
The latest release with the Devices tab fixes this for me 🎉

**Comment by nmqanh** [comment]
@jdelStrother would you mind showing us how to use it 👍

**Comment by jdelStrother** [comment]
@nmqanh : 

![image](https://cloud.githubusercontent.com/assets/2377/19773289/e5539a24-9c60-11e6-9ba0-a31744c17b74.png)

**Comment by cormacrelf** [comment]
I don't think that's what was requested. This only disables _modifying_ internal keyboard events, it doesn't suppress all key presses. Tick in devices panel = the modifications apply to this keyboard, no tick = the modifications do not apply. K-E doesn't yet have this functionality, UNLESS you remap every single key to none (what a pain!), and untick the external keyboard only.

**Comment by 0xABCDEF** [comment]
@jdelStrother It can indeed disable the external keyboard but still have **no effect** on internal keyboard.
<img width="743" alt="k-e" src="https://cloud.githubusercontent.com/assets/11064144/19826962/15d82dba-9dcc-11e6-9423-fe2f4f8e7e5f.png">

**Comment by nmqanh** [comment]
Hello everyone, I wrote a Node.js script to automatically disable internal keyboard when external keyboard connected and vice versa. It work directly with Karabiner-Elements without any hacking with mac using `csrutil disable`.

You can feel free to use my script here, I wrote a guide in README also: https://github.com/nmqanh/karabiner-elements-disable-internal-when-external-connected

**Comment by tekezo** [comment]
I've added this function in v0.90.63.
Please upgrade the latest version and configure in Devices tab.

![161106-0002](https://cloud.githubusercontent.com/assets/659178/20037881/ac8dedde-a46c-11e6-85d2-b5c7fd29238f.png)

**Comment by wez** [comment]
This is so much improved over the old implementation, since Karabiner can now configured not to disable the internal keyboard when my yubikey is attached.  Thank you!

**Comment by tedbillups** [comment]
18 months later... this disable internal keyboard feature just helped me so much with a broken keyboard issue on the MBP. Thank you @tekezo !

**Comment by rachiiemay** [comment]
v0.90.63 with the "Disable the built in keyboard..." function worked for me! Thank you so much!

**Comment by huangzonghao** [comment]
@tekezo  Is there a way that I can manually turn it on or off? Like executing a bash command? Thanks!

**Comment by mdeggies** [comment]
Awesome, thanks!! Screenshot shows the latest UI // go to Devices -> Advanced to disable

<img width="1018" alt="Screen Shot 2019-07-08 at 11 29 06 AM" src="https://user-images.githubusercontent.com/12939567/60833704-b1559e00-a173-11e9-86b3-a9e28f476611.png">

**Comment by bai** [comment]
Just saying: huge thanks for this feature 🙏 With a generation of MacBook Pros with butterfly keyboards that get broken so easily (even 2018/2019 ones!), this feature saved my day more than once.

**Comment by pwentz** [comment]
I'm not sure if anybody else is experiencing this, but I've found that this feature disables all the buttons on the internal keyboard _except_ for the power button to the top right. This makes this feature largely unusable for me because I set my keyboard on top of my internal one and the power button keeps getting incidentally tripped.

Is there any way around this problem? I tried to re-map the power key of my internal keyboard in karabinar, but that didn't seem to have any effect.
